File: cpp/03_advanced/src/Wall.cpp
#include "../include/Wall.h"

Wall::Wall(int x, int y, bool destructible) 
    : position(x, y), destructible(destructible) {
    
    if (destructible) {
        color = QColor(150, 100, 50);
    } else {
        color = QColor(100, 100, 120);
    }
}

File: csharp/03_advanced/SnakeGame/Models/Wall.cs
namespace SnakeGame.Models
{
    public class Wall
    {
        public int X { get; set; }
        public int Y { get; set; }
        public bool IsDestructible { get; set; }
        public string Color { get; set; }

        public Wall(int x, int y, bool destructible = false)
        {
            X = x;
            Y = y;
            IsDestructible = destructible;
            Color = destructible ? "#8B4513" : "#4A4A5A";
        }
    }
}
